src/lib/types.ts

```diff
@@ -72,6 +81,11 @@ export type SettingOption = {
   label: string
 }
 
+export type TSettingOption<T> = {
```

Comment:
Can we reuse the `SettingOption` type above? For all places where SettingOption is used you could use `string` for now or see what is uses and use that type.

Reply:
I saw you comment about the transition, but I think (if possible) we want to change it in this PR.

Reply:
Check my comment below.